airflow/models/dag.py

```diff
@@ -447,6 +453,7 @@ def __init__(
         owner_links: dict[str, str] | None = None,
         auto_register: bool = True,
         fail_stop: bool = False,
+        default_run_id: Callable | None = None,
```

Comment:
Can you add some type hints so that mypy can check the signature if the callable is customized?
```suggestion
        default_run_id: Callable[[DagRunType, DateTime, DataInterval, dict], str] | None = None,
```